packages/core/admin/admin/src/content-manager/components/FormInputs/Date.tsx

```diff
@@ -16,6 +16,7 @@ const DateInput = forwardRef<HTMLInputElement, InputProps>(
     const fieldRef = useFocusInputField(name);
 
     const composedRefs = useComposedRefs<HTMLInputElement | null>(ref, fieldRef);
+    const value = typeof field.value === 'string' ? new Date(field.value) : field.value;
```

Comment:
I think in the frontend we'll always prefer to deal with Dates, I don't see upsides for strings 🤔 So my intuition would be to transform it as early as possible after data fetching. 

>should we instead be transforming these values in the same place we transform the rest of the document?

what place are you referring to?

Reply:
We pipe the fetched document in the edit view here – https://github.com/strapi/strapi/blob/84b8d2ccd465193f6de5e9ae3592a39e25917a0d/packages/core/admin/admin/src/content-manager/pages/EditView/EditViewPage.tsx#L137 through a series of transformations using the schema and components, just stuff like setting up relations to be arrays & applying temp keys which we need for D&D etc.

I think the downside of transforming it at that level is that we don't have an "out", the nice thing of how the form is working at the moment is that the expected API format is in the form and you can just use the `useDocumentActions` hook to post it to the API in publish/create/update etc. I'm facing this issue in relations because the UI need to juggle the relations already connected etc.

I think maybe your solution is better because only the input cares about what value it's working with and the input/ouput is the same.... 🤔

Reply:
Ah I see. In that case I guess we can go with this current solution, since it's simpler, and see if it causes any issues elsewhere down the line